metadata-ingestion/setup.py

```diff
@@ -215,8 +215,8 @@ def get_long_description():
 
 iceberg_common = {
     # Iceberg Python SDK
-    "acryl-iceberg-legacy==0.0.4",
-    "azure-identity==1.10.0",
+    "pyiceberg==0.4.0",
+    "pyarrow>=9.0.0, <13.0.0",
```

Comment:
I don't have a hard reason to pin `pyiceberg`.  I did this at the time because pyiceberg is in constant evolution and I wanted to limit the trouble caused by an API change.  But after thinking about it, I think the section of the API that Datahub is using is pretty stable, so removing the pin should be fine.